target/linux/ipq806x/base-files/etc/board.d/01_leds

```diff
@@ -8,39 +8,40 @@
 board_config_update
 
 board=$(board_name)
+boardname="${board##*,}"
 
 case "$board" in
-c2600)
-	ucidef_set_led_usbport "usb1" "USB 1" "${board}:white:usb_2" "usb1-port1" "usb2-port1"
-	ucidef_set_led_usbport "usb2" "USB 2" "${board}:white:usb_4" "usb3-port1" "usb4-port1"
-	ucidef_set_led_switch "wan" "wan" "${board}:white:wan" "switch0" "0x20"
-	ucidef_set_led_switch "lan" "lan" "${board}:white:lan" "switch0" "0x1e"
+avm,fritzbox-4040)
+	ucidef_set_led_wlan "wlan" "WLAN" "${boardname}:green:wlan" "phy0tpt" "phy1tpt"
```

Comment:
You need to use a fixed boardname here. The LEDs are named `fritz4040:<colour>:<function>`.

Reply:
You're right. I already fixed it on another branch, but I don't wanna push the changes yet until I have addressed the BOARD_NAME variable problem.

I left the LEDs names hard-coded as you suggested (fritz4040:white:wan).
I imagine we can't change the device-tree because we would break the LEDs of existing users which already have the LEDs defined in the file /etc/config/system. Is that correct?

Reply:
Changing LED names is always problematic as it requires at least a script which migrates existing configurations. Since the fritzbox 4040 doesn't have yet a stable image, it might be possible to either change the compatible string or the LED names.

Lets see what @chunkeey thinks about it, since the fritzbox 4040 commit was from him.